modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/api/SessionImpl.java

```diff
@@ -163,7 +162,7 @@ public CompletableFuture<AsyncResultSet> executeAsync(@Nullable Transaction tran
             );
 
             result.whenComplete((rs, th) -> {
-                if (th instanceof SessionNotFoundException) {
+                if (th instanceof IgniteException && ((IgniteException) th).code() == SESSION_NOT_FOUND_ERR) {
```

Comment:
We may want to have an utility method for this in future I believe.

Reply:
Done